### DISCUSSION TITLE
Strange result from 1.0/catalog/products/

### DISCUSSION THREAD

**Discussion by armsby** [mkb79/Audible#338] [Q&A] [answered]
I am trying to get information for a book use "1.0/catalog/products/"
im asking in the "us" market place
this is what the client _request sends
```
Method: GET
URL: https://api.audible.com/1.0/catalog/products/
kwargs: {'params': {'num_results': 50, 'title': 'And Less Than Kind', 'response_groups': 'media,series,product_desc,product_extended_attrs,contributors,category_ladders,rating,product_attrs'}}
```

This give me 1 result,
if I run the same with curl
```
curl 'https://api.audible.com/1.0/catalog/products/?title=And%20Less%20Than%20Kind&num_results=50&response_groups=media,series,product_desc,product_extended_attrs,contributors,category_ladders,rating,product_attrs'
```

I get 2 results and it is of cause the one I dont get from the script that is the right one, do anyone have any suggestions to what

**Comment by mkb79** [comment, marked as answer]
I'll found the reason for this behavior. It's the authenticated request, which will result in a response of only 1 item.

If you disable authentication like:

```python
auth = Authenticator.from_file(auth_file)
with Client(auth, country_code="us") as client:
    r = client.get(
        "catalog/products",
        num_results=50,
        title="And Less Than Kind",
        response_groups="media,series,product_desc,product_extended_attrs,contributors,category_ladders,rating,product_attrs",
        auth=None  # disable auth for this request
    )
    print(r)
```

you'll got a response with 2 results.

**Reply by armsby** [reply]
Thank you so much that fixed it for me

**Reply by mkb79** [reply]
I would only be interested to know why the Audible server only sends 1 result if the request is authenticated.

**Reply by armsby** [reply]
if you run the same towards the UK market you only get 1 result, so last night I was looking at the market selector code, as I thought that the error was there :) 

Can Audible get confused sometimes?
I dont expect that is the answer, but when it happens I seam to get results from the UK marketplace

**Reply by mkb79** [reply]
Yeah, the audiobook
ASIN:1980051097 / TITLE:And Less Than Kind
seams to be only available on the US marketplace. I do not found any information about these book on UK or DE website. 

So every Audible marketplace offers different audiobooks. So selecting the right marketplace is important!

**Reply by armsby** [reply]
the problem is that I did, This is the code I use
```
    def get_audible_metadata(self, TITLE, AUTHOR, NARRATOR):
        auth = audible.Authenticator.from_file("authfile")
        
        with audible.Client(auth=auth, country_code="us") as client:
            client.switch_marketplace('us')
            print(client.marketplace)
            books = client.get(
                "1.0/catalog/products/",
                num_results=50,
                title=TITLE,
                #author=AUTHOR,
                #narrator=NARRATOR,
                response_groups=','.join([
                    'media',
                    'series',
                    'product_desc',
                    'product_extended_attrs',
                    'contributors',
                    'category_ladders',
                    'rating',
                    'product_attrs',
                ]),
            )
```
when testing so I was sure that it was not in the wrong marketplace.
and it did print `us` from the marketplace

**Reply by mkb79** [reply]
The issue is using an authenticated client. The server seams to filter the response in that case. 

FYI:
When you are instantiate the `Client` with `country_code="us"` you do not need to run `client.switch_marketplace('us')`.
